include/irr/core/math/aabb2dSIMDIntBase.h

```diff
@@ -4,67 +4,87 @@
 #include "irr/core/math/irrMath.h"
 #include "irr/core/memory/memory.h"
 #include "irr/core/alloc/AlignedBase.h"
-#include "C:\IrrlichtBAW\IrrlichtBAW\include\vectorSIMD.h"
+#include "../include/vectorSIMD.h"
 
 #include <iostream>
 
+#define DWN_CAST_THIS_PTR static_cast<const CRTP<VECTOR_TYPE>*>(this)
+#define SIGN_FLIP_MASK_XY	core::vector3du32_SIMD(0x80000000, 0x80000000, 0x00000000, 0x00000000)
+
 namespace irr { namespace core {
 
 //will modify this class so aabbSIMDf will be able do derive from it too
 template<template<typename> typename CRTP, typename VECTOR_TYPE>
 class aabbSIMDBase
 {
-	static_assert(	
-		std::is_same<VECTOR_TYPE, core::vector4di32_SIMD>::value ||
-		std::is_same<VECTOR_TYPE, core::vector4du32_SIMD>::value, 
+	static_assert(
+		std::is_same<VECTOR_TYPE, core::vector2di32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector2du32_SIMD>::value ||
+		std::is_same<VECTOR_TYPE, core::vector2df_SIMD>::value,
 		"assertion failed: cannot use this type");
 
 public:
+
 	inline void addPoint(const VECTOR_TYPE& other)
 	{
-		static_cast<CRTP<VECTOR_TYPE>*>(this)->addBox();
+		DWN_CAST_THIS_PTR->addBox();
 	}
 
 	inline VECTOR_TYPE getExtent() const
 	{
-		return getMaxEdge() - getMinEdge();
+		return DWN_CAST_THIS_PTR->getMaxEdge() - static_cast<const CRTP<VECTOR_TYPE>*>(this)->getMinEdge();
 	}
 
-	inline VECTOR_TYPE getMinEdge() const
+	//need operator/ for core::vector32_i<uint32_t> and core::vector32_i<uint32_t>
+	/*inline VECTOR_TYPE getCenter() const
 	{
-		VECTOR_TYPE minEdge = box;
-		minEdge.makeSafe2D();
-		return minEdge;
-	}
+		return static_cast<const CRTP<VECTOR_TYPE>*>(this)->getMaxEdge() - static_cast<const CRTP<VECTOR_TYPE>*>(this)->getMinEdge();
+	}*/
 
-	inline VECTOR_TYPE getMaxEdge() const
+	//! Resets the bounding box to a one-point box.
+	/** \param x X coord of the point.
+	\param y Y coord of the point. */
+	inline void reset(const VECTOR_TYPE& point)
 	{
-		VECTOR_TYPE maxEdge = box.zwzw();
-		maxEdge.makeSafe2D();
-		return maxEdge;
-	}
-
-	inline CRTP<VECTOR_TYPE> reset(const VECTOR_TYPE& point) const
-	{
-		box = point.xyxy();
-		return *this;
+		CRTP<VECTOR_TYPE> pointBox(point.xyxy());
+		internalBoxRepresentation= pointBox;
```

Comment:
now you need an `std::move`

Reply:
why not ordinary copy assigment? We are not moving anything

Reply:
you could move the temporary `pointBox` into `this`

now what you have is incompileable, but I can assume you meant to copy the internal rep of the temporary `pointBox` into `this->internal....` which will get optimized by the compiler only if you're lucky.